### DISCUSSION TITLE
Does Avalonia support mobile touch screen pinch to zoom and panning of images ? What control ?

### DISCUSSION THREAD

**Discussion by buzzware** [AvaloniaUI/Avalonia#16897] [Q&A]
I want to show a large image and allow the user to zoom in and out and pan like any normal mobile app image display.
What control do I show the image in to achieve this ?

**Comment by thevortexcloud** [comment]
There is the third party pan and zoom control which probably does exactly what you want.

https://github.com/wieslawsoltes/PanAndZoom

Otherwise you will have to make your own.

**Reply by LaurentInSeattle** [reply]
I have had some difficulties using the above mentioned control, particularly when handling constraints.
I ended up using my "good old" WPF technique: Placing 'my' zoom/pan-able content into a ViewBox inside a Grid nested into a ScrollViewer. 
Let the ScrollViewer pan by controlling its Offset property, either programmatically or via using mouse or touch gestures.
Let the Viewbox zoom and fit the content using the dimension properties of the grid, again via mouse wheel or pinch.

Very simple, conceptually works just like WPF. 
HTH
